includes/blocks/class-sensei-course-results-block.php

```diff
@@ -77,21 +77,37 @@ public function render_block( $attributes ) {
 			return $this->block_content;
 		}
 
-		$structure       = Sensei_Course_Structure::instance( $course_id )->get( 'view' );
-		$block_content   = [];
-		$block_content[] = '<section class="wp-block-sensei-lms-course-results sensei-block-wrapper">';
-		$block_content[] = $this->render_total_grade( $course_id );
-		$block_content[] = $this->render_course_title( $course_id );
-
+		$class_name        = Sensei_Block_Helpers::block_class_with_default_style( $attributes, [] );
+		$structure         = Sensei_Course_Structure::instance( $course_id )->get( 'view' );
+		$other_lessons     = array_count_values( array_column( $structure, 'type' ) );
+		$has_other_lessons = array_key_exists( 'lesson', $other_lessons ) && 0 < $other_lessons['lesson'];
+		$block_content     = [];
+		$block_content[]   = '<section class="wp-block-sensei-lms-course-results sensei-block-wrapper ' . $class_name . '">';
+		$block_content[]   = $this->render_total_grade( $course_id );
+		$block_content[]   = $this->render_course_title( $course_id, $structure, $attributes );
+
+		// Render modules and associated lessons.
 		foreach ( $structure as $item ) {
 			if ( 'module' === $item['type'] ) {
 				$block_content[] = $this->render_module( $item, $attributes );
 			}
+		}
+
+		// Render other lessons.
+		if ( $has_other_lessons ) {
```

Comment:
Would make sense to make one single `if` from `<ul>` until `</ul>`? Or is there some case where we loop through the structure without adding this `ul`?

Reply:
Don't think so. Fixed in https://github.com/Automattic/sensei/pull/4279/commits/adaab36fcfc7140faf97429b7c4575ec2bed45c6.